File: app/models/job_offer.rb
require 'sanitize_html'

class JobOffer < ApplicationRecord
  belongs_to :source, touch: true
  validates :url, :source, presence: true, allow_blank: false
  validates :url, uniqueness: { scope: :source_id }

  def self.from_entry(source_entry)
    new(source_entry.to_h)
  end

  def content
    self[:content] ||= SanitizeHTML.(raw_content)
  end

  def to_s
    "(#{updated_at}) #{title}"
  end

  def duplicate?
    source.job_offers.where(url: url).any?
  end
end

File: client/app/components/JobOffer.js
import React from 'react';

const JobOffer = (jobOffer) => {
  return (
    <div>
      <h2>
        <a href={jobOffer.url}>{ jobOffer.title }</a>
      </h2>
      <p>{ jobOffer.content }</p>
    </div>
  );
};

export default JobOffer;
